File: share-code/bai-tap-lap-trinh/python/comprehensive/76.py
import re

input = "2c+)1o!@(*&!y@&chi&^c!@#o3co$23%%^nd$#$!uo5ng"

str_to_decode = re.sub("[^A-Za-z]", "", input)
key           = re.sub("[^0-9]+", "", input)

start_index = 0
result      = ""

for i, current in enumerate(key):
    end_index   = start_index + int(current)
    word        = str_to_decode[start_index:end_index].capitalize() + " "
    result      += word
    start_index = end_index

print(result.strip())

File: share-code/bai-tap-lap-trinh/cpp/09-tonghop/76.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
    string input = "2c+)1o!@(*&!y@&chi&^c!@#o3co$23%%^nd$#$!uo5ng";
    string str;
    string key;
    string result = "";
    int startIndex = 0;

    for (char c : input)
    {
        if (isdigit(c))
        {
            key += c;
        }
        else if (isalpha(c))
        {
            str += c;
        }
    }

    for (int i = 0; i < key.length(); i++)
    {
        int size = key[i] - '0';
        string word = str.substr(startIndex, size) + " ";
        word[0] = toupper(word[0]);
        result += word;
        startIndex += size;
    }

    cout << result << endl;
    return 0;
}